app/parsers/bulkrax/csv_parser.rb

```diff
@@ -128,10 +129,13 @@ def current_work_ids
       when 'worktype'
         ActiveFedora::SolrService.query("has_model_ssim:#{importerexporter.export_source + extra_filters}", rows: 2_000_000_000).map(&:id)
       when 'importer'
-        importer = Bulkrax::Importer.find(importerexporter.export_source)
-        # TODO: this seems likely to be broken
-        entry_ids = importer.entries.where(type: importer.parser.entry_class.to_s, last_error: [nil, {}, '']).map(&:identifier)
-        ActiveFedora::SolrService.query("#{work_identifier}_tesim:(#{entry_ids.join(' OR ')})#{extra_filters}", rows: 2_000_000_000).map(&:id)
+        entry_ids = importer.entries.pluck(:id)
```

Comment:
this is 3 queries (one for initial entries, one for statuses and a last one of the entries of those statuses) how ever it is only every 3 queries and not an n+1. I'd rather do it in 1 (entries where that last status is "Complete" and the importer is this importer), but I couldn't figure out how to do that.